### PR TITLE
Replace slime extract experiments, required for biotech, with dissection surgery

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Adds back dissection surgery for the purposes of performing experiments. Replaces the existing slime extract fetch quests with dissection surgery for biotech, and adds xenomorph dissection as an extra discount (though I would like to see a node require it some day).

## Why It's Good For The Game

Science has no incentive to actually scan slime extracts for biotech, as the primary consumer is medical.

In practice, this means that the grey slime extract experiment, which can be done *round-start*, is frequently left unperformed. This lets medical have some level of control over their research, while still leaving the actual act of researching the node into the hands of the science department, or heads of staff.

## Changelog

:cl:
add: Added back dissection surgery, which can be used with an operating computer to perform experiments for research.
add: Biological technology's requirement has been changed from scanning one grey slime extract to performing a dissection surgery.
add: Advanced biotech's requirement has been changed from scanning three slime extracts to performing a dissection surgery on a nonhuman.
add: Xeno-organ biology and Alien Biological Tools' now have additional discounts for performing a dissection surgery on a xenomorph.
/:cl:

### CODE REVIEW

**Comment by nianjiilical** [comment]
does the dissection surgery actually kill the target? i think this is a good change overall but now that i think about it im not actually sure how accessible nonhuman corpses are to medbay without tapping into xenobio mutation juice or killing a crewmember

unless monkeys count in which case nevermind

**Comment by Mothblocks** [comment]
Dissection surgery requires a dead body, but does not kill you.

Monkeys work for the nonhuman experiment, and are even mentioned in the description for it as a suggestion.

**Comment by Burgermancoding** [comment]
image

File: code/modules/experisci/experiment/handlers/experiment_handler.dm [hunk 1]
```diff
@@ -56,6 +56,8 @@
 		RegisterSignal(parent, COMSIG_DOPPLER_ARRAY_EXPLOSION_DETECTED, .proc/try_run_doppler_experiment)
 	if(istype(parent, /obj/machinery/destructive_scanner))
 		RegisterSignal(parent, COMSIG_MACHINERY_DESTRUCTIVE_SCAN, .proc/try_run_destructive_experiment)
+	if(istype(parent, /obj/machinery/computer/operating))
```
Comment (Mothblocks) [inline code/modules/experisci/experiment/handlers/experiment_handler.dm:59]: For the record, this feels like a really weird God component. Project for another day.

File: code/modules/mob/mob_helpers.dm [hunk 2]
```diff
@@ -502,15 +502,6 @@
  * I wish examine was less copypasted. (oranges say, be the change you want to see buddy)
  */
 /mob/proc/common_trait_examine()
-	if(HAS_TRAIT(src, TRAIT_DISSECTED))
```
Comment (Mothblocks) [inline code/modules/mob/mob_helpers.dm:505]: This was all just leftover that I never removed from healing skill.

**Comment by Iplayforss13** [comment]
Dissection on xenomorph? I'd rather my alien tools not locked behind a round ending antagonist, And its no guarantee that xenos spawn with their 10 threat,

Make the xenomorph egg in xenobio spawn more often, and i'll be happy

**Comment by Orchian** [comment]
Good stuff

**Comment by moocowswag** [comment]
Geneticists and roboticists actually do have an incentive to research bio tech for public gene consoles and posi brains. (Esp traitor roboticists who want to make/subvert an ai)

And their ability to actually get to xenobio and scan an extract varies wildly depending on the map (from just having to screw driver a non-reinforced glass window to having to hack thru 3 airlocks, 2 of which are cycled/bolted at round start)

This pr will at least allow roboticists to more reliably unlock bio tech without having to go far.....far... out of their way to do so. (With the main hurdle requiring a humanoid which they get from a single person wanting to be borged or via genetics)

The last role I could see being likely to doing the dissection surgery is a medical borg which are also reliant on bio tech in order to eventually unlock their upgrades.

**Comment by EricZilla** [comment]
> Dissection on xenomorph? I'd rather my alien tools not locked behind a round ending antagonist, And its no guarantee that xenos spawn with their 10 threat,
> 
> Make the xenomorph egg in xenobio spawn more often, and i'll be happy

Its a discount to the research, not a prerequisite

**Comment by softcerv** [comment]
> > Dissection on xenomorph? I'd rather my alien tools not locked behind a round ending antagonist, And its no guarantee that xenos spawn with their 10 threat,
> > Make the xenomorph egg in xenobio spawn more often, and i'll be happy
> 
> Its a discount to the research, not a prerequisite

I thought that the slime experiment was a prerequisite to the alien medical tools, but I could be wrong.

**Comment by Mothblocks** [comment]
Did, did you mean to use two accounts

**Comment by Mothblocks** [comment]
Oh no I misread

Anyway, other way around. alien dissection is a discount to xeno-organ bio, only used with limb growers.

Might make it a discount on the other as well

**Comment by softcerv** [comment]
It'd probably be a good idea to make it a discount instead of a requirment for the one that unlocks the alien medical tools, otherwise you'd probably never see it researched much.

**Comment by softcerv** [comment]
Does this happen to work on non-sentient alien corpses (like the ones from lavaland)

**Comment by Mothblocks** [comment]
I think those are simple mobs, not carbons

**Comment by Nexofas** [comment]
Good PR

**Comment by Mothblocks** [comment]
The xenomorph dissections are now discounts rather than requirements, and the original slime experiments still exist exclusively for those.

In the future I'd like for xenomorph dissection to be required for *something*, but not when the other stuff already has some weird requirements (like finding alien tools and deconstructing them).

**Comment by TheBonded** [comment]
> In the future I'd like for xenomorph dissection to be required for _something_, but not when the other stuff already has some weird requirements (like finding alien tools and deconstructing them).

perhaps it could allow you to make a limb grower disk that (at an extremely expensive chemical cost, like strange reagent + other hard to get things) allows you to make abductor organs, or something of the sort. Perhaps even other more exotic organs, like hive nodes and the golem adamantine resonators and tongues.

**Comment by datorangebottle** [comment]
any chance that you could enable the dissection surgery discount for xeno tech be discounted by dissecting an abductor corpse OR a xenomorph corpse, instead of just xenomorph? You get a free abductor corpse with the alien tools you find in mining. it'd make sense.

**Comment by Mothblocks** [comment]
Current ExperiSci code doesn't allow for alternative experiments like that, so out of scope for now

**Comment by Draggeru** [comment]
As both a med main and a science main this gives me happiness noises all around

**Comment by optimumtact** [comment]
no thanks depts are meant to interoperate

**Comment by carshalash** [comment]
But.. doesn't this make it so ONLY science is doing experiments again?

**Pull request merged**